app/controllers/SelfAssessmentController.scala

```diff
@@ -106,4 +111,20 @@ class SelfAssessmentController @Inject()(
       }
     }
 
+  def viewPayments: Action[AnyContent] =
+    authJourney.authWithSelfAssessment.async { implicit request =>
+      request.saUserType match {
+        case ActivatedOnlineFilerSelfAssessmentUser(saUtr) =>
+          (for {
```

Comment:
Simplify this to use map. You should look to only use for yield when there is multiple futures to compute as for yield comes with additional steps